### DISCUSSION TITLE
Webgazer: Calibrating a specific location on screen

### DISCUSSION THREAD

**Discussion by armin-maddah** [jspsych/jsPsych#2624] [Q&A]
Hello, I am new to experimenting with webgazer extension. Unfortunately not much documentation on use of the extension functions, so I had to head to here finally and ask for help!

I am trying to register a specific location on the screen for calibration. I need this since I want to track gaze on specific parts of my stimulus. My stimulus has multiple parts and multiple patterns which should be used for a categorization response. Before the actual trial begins, I want participants to pay attention to the part of the screen that is going to be indicative about the category of the stimulus. Here's a code I have for a trial before the actual task begins. When I added this to my timeline and then checked the last recorded data (via jsPsych.data.get().last() in console) I can not see any webgazer related data.

```
var pants_calibrate = {
    type: jsPsychHtmlButtonResponse,
    stimulus: '<p>Click on the button to begin.</p>',
    choices: ['PANTS'],
    button_html: '<button class="calibrate-btn-pants">%choice%</button>',
    on_start: function(pants_calibrate){
        jsPsych.extensions.webgazer.startMouseCalibration()
    },
    on_finish: function(){
        jsPsych.extensions.webgazer.stopMouseCalibration()
    }
}
```
I am using a separate CSS to manipulate the location of the button for different sort of trials. Would appreciate any help! Most probably I am not doing something right here! Thank you!

**Comment by jodeleeuw** [comment]
Hi @armin-neuro 

If you want to gather webgazer data during a trial you'll need to turn on tracking for that trial. The [eye tracking overview](https://www.jspsych.org/7.2/overview/eye-tracking/#adding-eye-tracking-to-a-trial) describes how to do this.

**Reply by armin-maddah** [reply]
Hi @jodeleeuw !
Thank you for the quick response!
How about the rest of the code? would then this single trial be used for calibration? all in all, do you have any recommendations for the case I am explaining? My results are unfortunately not good/accurate. Do you think we can use webgazer's data for distinguishing between gaze on different parts of a single stimulus? compared to the case of trials where several separate stimuli are presented. Thanks again for your support!

**Reply by jodeleeuw** [reply]
> How about the rest of the code? would then this single trial be used for calibration? 

The [eye tracking overview page](https://www.jspsych.org/7.2/overview/eye-tracking/) covers how to perform calibration with the webgazer-calibrate plugin. You can customize the calibration pretty much however you want, including using multiple trials for calibration throughout the experiment.

> My results are unfortunately not good/accurate. Do you think we can use webgazer's data for distinguishing between gaze on different parts of a single stimulus?

That depends on how big the stimulus is on the screen. How big are the stimuli?
